irods/manager/data_object_manager.py

```diff
@@ -84,29 +84,44 @@ def should_parallelize_transfer( self,
                                      num_threads = 0,
                                      obj_sz = 1+MAXIMUM_SINGLE_THREADED_TRANSFER_SIZE,
                                      server_version_hint = (),
-                                     measured_obj_size = ()  ## output variable. If a list is provided, it shall
-                                                              # be truncated to contain one value, the size of the
-                                                              # seekable object (if one is provided for `obj_sz').
+                                     measured_obj_size = (), ## output variable. If a list is provided, it shall
+                                                              # be truncated to contain one value, the presumed size of the
+                                                              # file or data object - if one is provided for `obj_sz'.  That size is
+                                                              # determined normally by stat'ing (using a seek-to-end operation) but
+                                                              # can be overridden using the DATA_SIZE_KW option.
+                                     open_options = ()        # If this parameter is of type dict, set the DATA_SIZE_KW item from
+                                                              # the stat'ed size
         ):
 
+        provided_data_size = int(dict(open_options).get(kw.DATA_SIZE_KW,"-1"))
         # Allow an environment variable to override the detection of the server version.
         # Example: $ export IRODS_VERSION_OVERRIDE="4,2,9" ;  python -m irods.parallel ...
         server_version = ( ast.literal_eval(os.environ.get('IRODS_VERSION_OVERRIDE', '()' )) or server_version_hint or 
                            self.server_version )
-        if num_threads == 1 or ( server_version < parallel.MINIMUM_SERVER_VERSION ):
-            return False
-        if getattr(obj_sz,'seek',None) :
-            pos = obj_sz.tell()
-            size = obj_sz.seek(0,os.SEEK_END)
-            if not isinstance(size,six.integer_types):
-                size = obj_sz.tell()
-            obj_sz.seek(pos,os.SEEK_SET)
-            if isinstance(measured_obj_size,list): measured_obj_size[:] = [size]
-        else:
-            size = obj_sz
-            assert (size > -1)
-        return size > MAXIMUM_SINGLE_THREADED_TRANSFER_SIZE
-
+        size = None
+        try:
+            if num_threads == 1 or ( server_version < parallel.MINIMUM_SERVER_VERSION ):
+                return False
+            if getattr(obj_sz,'seek',None) :
```

Comment:
Looks like this code path does not result in any `return` statement

Reply:
> Looks like this code path does not result in any `return` statement

Good catch, yep. I think the situation has been corrected.